### PR TITLE
Skip Dialog interaction on macos

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

This test has started consistently flaking on the mac luci bots. It is unclear what the cause is currently.

## Related Issues

https://github.com/flutter/flutter/issues/29384

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jonahwilliams** [comment]
We should land this immediately to reduce the current flakes

**Comment by dnfield** [comment]
LGTM

Should we just disable this on all platforms until we can figure out why it's flaky?

File: packages/flutter/test/material/text_field_focus_test.dart [hunk 1]
```diff
@@ -130,7 +132,7 @@ void main() {
     await tester.pumpWidget(Container());
 
     expect(tester.testTextInput.isVisible, isFalse);
-  });
+  }, skip: Platform.isMacOS);
```
Comment (dnfield) [inline packages/flutter/test/material/text_field_focus_test.dart]: `skip: true` until we deflake it?

Comment (jonahwilliams) [reply packages/flutter/test/material/text_field_focus_test.dart]: Sounds good, added a link to the bug too

**Comment by jonahwilliams** [comment]
Landing on red to fix build

**Pull request merged**